pkg/kubelet/certificate/certificate_manager.go

```diff
@@ -46,85 +46,142 @@ const (
 	Start()
-	// GetCertificate gets the current certificate from the certificate
-	// manager. This function matches the signature required by
-	// tls.Config.GetCertificate so it can be passed as TLS configuration. A
-	// TLS server will automatically call back here to get the correct
-	// certificate when establishing each new connection.
-	GetCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error)
+	// Current returns the currently selected certificate from the
+	// certificate manager, as well as the associated certificate and key data
+	// in PEM format.
+	Current() *CertKeyData
+}
+
+type CertKeyData struct {
+	Certificate *tls.Certificate
+	CertPEM     []byte
+	KeyPEM      []byte
+}
+
+// Config is the set of configuration parameters available for a new Manager.
+type Config struct {
+	// CertificateSigningRequestClient will be used for signing new certificate
+	// requests generated when a key rotation occurs. It must be set either at
+	// initialization or by using CertificateSigningRequestClient before
+	// Manager.Start() is called.
+	CertificateSigningRequestClient certificatesclient.CertificateSigningRequestInterface
+	// Template is the CertificateRequest that will be used as a template for
+	// generating certificate signing requests for all new keys generated as
+	// part of rotation. It follows the same rules as the template parameter of
+	// crypto.x509.CreateCertificateRequest in the Go standard libraries.
+	Template *x509.CertificateRequest
+	// Usages is the types of usages that certificates generated by the manager
+	// can be used for.
+	Usages []certificates.KeyUsage
+	// CertificateStore is a persistent store where the current cert/key is
+	// kept and future cert/key pairs will be persisted after they are
+	// generated.
+	CertificateStore Store
+	// CertificateRotationPercent is the percentage of time remaining of the
+	// current certificate's lifetime before it will be rotated. If this is set
+	// to 100 (or higher) the certificate will be rotated each time it is
+	// checked. If this is set to 0, it will never be rotated.
+	CertificateRotationPercent uint
+	// Rotation is a call back function which, if provided, will be invoked
+	// after a rotation has occurred.
+	Rotation func()
+	// BootstrapCertificatePEM is the certificate data that will be used to
+	// authenticate connections with the CertificateSigningRequestClient and
+	// request a new certificate, if there is no current certificate available
+	// in the CertificateStore. If there is a current certificate available,
+	// this will be ignored.
+	BootstrapCertificatePEM []byte
+	// BootstrapKeyPEM is the key data that will be used to authenticate
+	// connections with the CertificateSigningRequestClient and request a new
+	// certificate, if there is no current certificate available in the
+	// CertificateStore. If there is a current certificate available, this will
+	// be ignored.
+	BootstrapKeyPEM []byte
```

Comment:
Well, you can use any authn to initialize your TLS client, but initializing your TLS client doesn't happen in the certificate manager. The certificate manager is the consumer of an already initialized client, only responsible for rotating as expiration approaches and returning the current cert/key pair. If you aren't going to use certs and keys, you wouldn't be using the CertificateManager, which is only designed for managing cert/key pairs. 

There is already a `CertificateSigningRequestInterface` in this struct, it's the first member.

I see the comments for these two members are incorrect and may have mislead you. I'll update it. The bootstrap cert/key pair are the data returned from the certificate manager if the `Store` doesn't currently have any cert/key pairs to return.